pyriemann_qiskit/classification.py

```diff
@@ -0,0 +1,381 @@
+
+    See Also
+    --------
+    QuanticSVM
+    QuanticVQC
+
+    """
+
+    def __init__(self, target, quantum=True, q_account_token=None,
+                 process_vector=lambda v: v, verbose=True,
+                 test_input={}, test_per=0.33):
+        self.verbose = verbose
+        self._log("Initializing Quantum Classifier")
+        self.test_input = test_input
+        self.test_per = test_per
+        self.process_vector = process_vector
+        self.q_account_token = q_account_token
+        self.target = target
+        self.quantum = quantum
+        # protected field for child classes
+        self._training_input = {}
+
+    def _init_quantum(self):
+        if self.quantum:
+            aqua_globals.random_seed = datetime.now().microsecond
+            self._log("seed = ", aqua_globals.random_seed)
+            if self.q_account_token:
+                self._log("Real quantum computation will be performed")
+                IBMQ.delete_account()
+                IBMQ.save_account(self.q_account_token)
+                IBMQ.load_account()
+                self._log("Getting provider...")
+                self._provider = IBMQ.get_provider(hub='ibm-q')
+            else:
+                self._log("Quantum simulation will be performed")
+                self._backend = BasicAer.get_backend('qasm_simulator')
+        else:
+            self._log("Classical SVM will be performed")
+
+    def _log(self, *values):
+        if self.verbose:
+            print("[QClass] ", *values)
+
+    def _vectorize(self, X):
+        vector = X.reshape(len(X), self._feature_dim)
+        return [self.process_vector(x) for x in vector]
+
+    def _split_target_and_non_target(self, X, y):
+        self._log("""[Warning] Spitting target from non target.
+                 Only binary classification is supported.""")
+        first_matrix_in_X = np.atleast_2d(X[0])
+        n_matrices, n_channels = first_matrix_in_X.shape[:2]
+        self._feature_dim = n_channels * n_matrices
+        self._log("Feature dimension = ", self._feature_dim)
+        Xta = X[y == self.target]
+        Xnt = X[np.logical_not(y == self.target)]
+        vect_Xta = self._vectorize(Xta)
+        vect_Xnt = self._vectorize(Xnt)
+        self._new_feature_dim = len(vect_Xta[0])
+        self._log("Feature dimension after vector processing = ",
+                  self._new_feature_dim)
+        return (vect_Xta, vect_Xnt)
+
+    def _additional_setup(self):
+        self._log("There is no additional setup.")
+
+    def fit(self, X, y):
+        """Prepare the training data and the quantum backend
+
+        Parameters
+        ----------
+        X : ndarray, shape (n_trials, n_channels, n_channels) | \
+                shape (n_trials, n_channels * n_channels)
+            ndarray of SPD matrices. Matrices can be provided in
+            raw format (3D) or already vectorized (2D) such as after projection
+            into the tangent space. Raw matrices are naively inlined.
```

Comment:
Due to the symmetry of SPD matrices, naive vectorization gives a redundant output (halft of the vector is a copy of the other part), of length `n_channels * n_channels`. It exposes your classifier to the curse of dimensionality. If you keep only the upper triangular part, your output will be `n_channels * (n_channels + 1) / 2`, and your classifier will be more robust.

But, I think that, for a generic code, vectorization must be made outside of `QuanticClassifierBase`, using a scikit-learn pipeline (and in this pipeline, you can use pyRiemann merthods to vectorize covariance matrices).
Hence, `QuanticClassifierBase` will process only 2D inputs, of shape (n_trials, n_features), making no implicite hypothesis on the nature of inputs.

Reply:
In general, the `process_vector` attribute can cover this use case, by providing something like this:
![image](https://user-images.githubusercontent.com/6229031/135672568-01f6057e-608c-4cbf-8b93-9270be08ab19.png)

But I can see your point. Better to let pyRiemann taking care of this, it will be more error proning.